modules/RTC/RTCUtils.js

```diff
@@ -749,7 +763,49 @@ class RTCUtils extends Listenable {
         initRawEnumerateDevicesWithCallback();
 
         return new Promise((resolve, reject) => {
-            if (RTCBrowserType.isFirefox()) {
+            if (RTCBrowserType.usesNewGumFlow()) {
+                this.RTCPeerConnectionType = window.RTCPeerConnection;
+
+                this.getUserMedia
+                    = (constraints, successCallback, errorCallback) =>
+                        window.navigator.mediaDevices
+                            .getUserMedia(constraints)
+                            .then(stream => {
+                                successCallback && successCallback(stream);
+
+                                return stream;
+                            })
+                            .catch(err => {
+                                errorCallback && errorCallback(err);
+
+                                return Promise.reject(err);
+                            });
+
+                this.enumerateDevices = callback =>
```

Comment:
same as above

Reply:
enumerateDevices looks to always return a promise in the getusermedia.js shim. I don't see it looking for a callback and calling it.